engines/dradis-api/app/controllers/dradis/ce/api/v3/upload_controller.rb

```diff
@@ -0,0 +1,48 @@
+module Dradis::CE::API
+  module V3
+    class UploadController < Dradis::CE::API::APIController
+      include Dradis::CE::API::ProjectScoped
+      include Uploaded
+
+      skip_before_action :json_required, only: [:create]
+
+      def create
+        filename = CGI::escape upload_params[:file].original_filename
+        # add the file as an attachment
+        attachment = Attachment.new(filename, node_id: current_project.plugin_uploads_node.id)
+        attachment << upload_params[:file].read
+        attachment.save
+
+        if Rails.env.production? || (File.size(attachment.fullpath) > 1024 * 1024)
+          @job_id = process_upload_background(attachment: attachment).job_id
+          @status = :queued
+        else
+          process_upload_inline(attachment: attachment)
+          @status = :completed
+        end
+      end
+
+      def show
+        tracker = JobTracker.new(job_id: params[:job_id], queue_name: 'dradis_upload')
```

Comment:
- this is vulnerable to :ID enumeration
- the hard coded queue name is a smell. We should probably go through UploadedJob :queue_as (not sure how, but there should be a way to retrieve it).